OsmAnd/src/net/osmand/plus/helpers/GpsFilterHelper.java

```diff
@@ -0,0 +1,755 @@
+package net.osmand.plus.helpers;
+
+import android.graphics.Typeface;
+import android.os.AsyncTask;
+import android.text.SpannableString;
+import android.text.Spanned;
+import android.text.style.ForegroundColorSpan;
+import android.text.style.StyleSpan;
+
+import net.osmand.GPXUtilities.GPXFile;
+import net.osmand.GPXUtilities.GPXTrackAnalysis;
+import net.osmand.GPXUtilities.Track;
+import net.osmand.GPXUtilities.TrkSegment;
+import net.osmand.GPXUtilities.WptPt;
+import net.osmand.plus.ColorUtilities;
+import net.osmand.plus.GpxSelectionHelper.SelectedGpxFile;
+import net.osmand.plus.OsmAndFormatter;
+import net.osmand.plus.OsmandApplication;
+import net.osmand.plus.R;
+import net.osmand.plus.Version;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.StringRes;
+
+public class GpsFilterHelper {
```

Comment:
GpsFilterHelper should filter and not hold state of single filtered gpx.
Suggestion: create child of SelectedGpxFile like FilteredSelectedGpxFile and store it in original SelectedGpxFile. leftPoints, totalPoints, filters, etc. should be in FilteredSelectedGpxFile.